cds-snc/platform-forms-client
# Formatted Date component

# Summary | Résumé

Adds a Formatted date component.

### Things to test
- Add a date element
- Select different formats
- Ensure date inputs display in correct format on Preview/Form view
- In multi-page forms, ensure dates are displayed correctly according to selected format on Review
- Ensure dates are displayed correctly when downloading responses in any format

### Note
While the old date "component" (actually just a templated input) has been removed from the AddElement dialog, the rest of the code to support it remains in place so that existing forms or form files that include that type of input will continue to work as-is. There is no conversion/migration to the new component.


| Add element dialog                                          | Element panel                                        |
| ----------------------------------------------- | -------------------------------------------- |
| <img width="805" alt="image" src="https://github.com/user-attachments/assets/bac284a3-99a2-4ab6-acaf-342740d1ba52"> | <img width="807" alt="image" src="https://github.com/user-attachments/assets/1503a444-77ca-48b2-a592-800c94358b93"> |

| More modal/Options                                         | Form/Preview render                                       |
| ----------------------------------------------- | -------------------------------------------- |
| <img width="782" alt="image" src="https://github.com/user-attachments/assets/db5c0387-17b8-4d1f-81ea-d292e1eae302"> | <img width="301" alt="image" src="https://github.com/user-attachments/assets/62700f35-60e4-4cf4-a48d-6b3ae00eacd3"> |
